GW2EIEvtcParser/EncounterLogic/Convergences/MountBalriorConvergenceInstance.cs

```diff
@@ -0,0 +1,157 @@
+        phases[0].AddTargets(Targets, log);
+        SingleActor target = Targets.FirstOrDefault(x => x.IsAnySpecies([TargetID.DecimaTheStormsingerConv, TargetID.GreerTheBlightbringerConv, TargetID.UraTheSteamshriekerConv])) ?? throw new MissingKeyActorsException("Decima / Greer / Ura not found");
+        IReadOnlyList<Segment> hpUpdates = target.GetHealthUpdates(log);
+
+        // Full Fight Phase
+        string phaseName = "";
+        switch (target.ID)
+        {
+            case (int)TargetID.DecimaTheStormsingerConv:
+                phaseName = "Full Decima";
+                break;
+            case (int)TargetID.GreerTheBlightbringerConv:
+                phaseName = "Full Greer";
+                break;
+            case (int)TargetID.UraTheSteamshriekerConv:
+                phaseName = "Full Ura";
+                break;
+        }
+        var fullPhase = new PhaseData(Math.Max(log.FightData.FightStart, target.FirstAware), Math.Min(target.LastAware, log.FightData.FightEnd), phaseName);
+        fullPhase.AddTarget(target, log);
+        phases.Add(fullPhase);
+
+        // Sub Phases
+        Segment start = hpUpdates.FirstOrDefault(x => x.Value <= 100.0 && x.Value != 0 && x.Start != 0);
+        Segment end75 = hpUpdates.FirstOrDefault(x => x.Value < 75.0 && x.Value != 0);
+        Segment start75 = hpUpdates.FirstOrDefault(x => x.Value < 75.0 && x.Value != 0 && x.Start > end75.End);
+        Segment end50 = hpUpdates.FirstOrDefault(x => x.Value < 50.0 && x.Value != 0);
+        Segment start50 = hpUpdates.FirstOrDefault(x => x.Value < 50.0 && x.Value != 0 && x.Start > end50.End);
+        Segment end25 = hpUpdates.FirstOrDefault(x => x.Value < 25.0 && x.Value != 0);
+        Segment final = hpUpdates.FirstOrDefault(x => x.Value < 25.0 && x.Start > end25.End);
+
+        // 100-75, Warclaw, 75-50, Warclaw, 50-25, Warclaw, 25-0
+        PhaseData phase1 = new(start.Start, Math.Min(end75.Start, log.FightData.FightEnd), "Phase 1");
+        PhaseData phase2 = new(start75.Start, Math.Min(end50.Start, log.FightData.FightEnd), "Phase 2");
+        PhaseData phase3 = new(start50.Start, Math.Min(end25.Start, log.FightData.FightEnd), "Phase 3");
+        PhaseData phase4 = new(final.Start, Math.Min(target.AgentItem.LastAware, log.FightData.FightEnd), "Phase 4");
+        PhaseData warclaw1 = new(end75.Start, Math.Min(start75.Start, log.FightData.FightEnd), "Warclaw 1");
+        PhaseData warclaw2 = new(end50.Start, Math.Min(start50.Start, log.FightData.FightEnd), "Warclaw 2");
+        PhaseData warclaw3 = new(end25.Start, Math.Min(final.Start, log.FightData.FightEnd), "Warclaw 3");
+
+        phase1.AddTarget(target, log);
+        phase2.AddTarget(target, log);
+        phase3.AddTarget(target, log);
+        phase4.AddTarget(target, log);
+        warclaw1.AddTarget(target, log);
+        warclaw2.AddTarget(target, log);
+        warclaw3.AddTarget(target, log);
+        phase1.AddParentPhase(fullPhase);
```

Comment:
You have .WithParentPhase API on PhaseData, you can reduce code size a little bit by using new(start, end, name).WithParentPhase(fullPhase). Same for Nayos.